phoenix-core/src/main/java/org/apache/phoenix/coprocessor/GroupedAggregateRegionObserver.java

```diff
@@ -423,7 +426,14 @@ private RegionScanner scanUnordered(ObserverContext<RegionCoprocessorEnvironment
                         }
                     } while (hasMore && groupByCache.size() < limit);
                 }
-            } finally {
+            }  catch(NotServingRegionException e){
+                if(ScanUtil.isLocalIndex(scan)) {
+                    Exception cause = new StaleRegionBoundaryCacheException(c.getEnvironment().getRegion().getRegionInfo().getTable().getNameAsString());
+                    throw new DoNotRetryIOException(cause.getMessage(), cause);
+                } else {
+                    throw e;
```

Comment:
There is a chance of split can happen after preScannerOpen and before postScannerOpen. I think we can throw stale region boundary exception all the cases because even when we throw NSRE we again need to open the scanner which checks the region boundaries and throw the stale region boundary exception any way. Will check it once.